src/RedemptionPool.sol

```diff
@@ -98,74 +140,98 @@ contract RedemptionPool is Ownable {
+        _userGROBalance[msg.sender] += _amount;
         // Increases the total deposited by the amount
         totalGRO += _amount;
-        // Emits the Deposit event
-        emit Deposit(msg.sender, _amount);
     }
 
     /// @notice withdraw deposited GRO tokens before the deadline
     /// @param _amount amount of GRO tokens to withdraw
-    function withdraw(uint256 _amount) external {
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] > _amount) {
-            revert RedemptionErrors.UserBalanceToSmall();
-        }
-        _userBalance[msg.sender] -= _amount;
+    function withdraw(uint256 _amount) external onlyBeforeDeadline {
+        if (_userGROBalance[msg.sender] < _amount)
+            revert RedemptionErrors.InsufficientBalance();
+
+        _userGROBalance[msg.sender] -= _amount;
         totalGRO -= _amount;
         GRO.safeTransfer(msg.sender, _amount);
         emit Withdraw(msg.sender, _amount);
     }
 
-    /// @notice Allow to withdraw cUSDC based on amount of GRO tokens deposited per user
-    function claim() external {
-        if (block.timestamp <= DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] <= 0) {
-            revert RedemptionErrors.NoUserBalance();
-        }
+    /**
+     * @notice Allow users to claim their share of USDC tokens based on the amount of GRO tokens they have deposited
+     * @dev Users must have a positive GRO token balance and a non-zero claim available to make a claim
+     * @dev The deadline for making GRO deposits must have passed
+     * @dev Redeems the user's cUSDC tokens for an equivalent amount of USDC tokens and transfers them to the user's address
+     * @dev Decreases the user's claims and contract accounting by the amount claimed
+     */
+    function claim(uint256 _amount) external onlyAfterDeadline {
+        // Get the amount of cUSDC tokens available for the user to claim
         uint256 userClaim = getSharesAvailable(msg.sender);
-        if (userClaim == 0) {
-            revert RedemptionErrors.NoUserClaim();
-        }
-        _userClaims[msg.sender] += userClaim;
-        IERC20(CUSDC).transfer(msg.sender, userClaim);
-        emit Claim(msg.sender, userClaim);
+
+        // Cap the user's claim to the available balance of cUSDC tokens
+        if (_amount > userClaim) _amount = userClaim;
+
+        // Revert if nothing to claim
+        if (_amount == 0) revert RedemptionErrors.InsufficientBalance();
+
+        // Redeem the user's cUSDC tokens for USDC tokens
+        // and transfer the USDC tokens to the user's address
+        uint256 redeemResult = ICERC20(CUSDC).redeem(_amount);
+        if (redeemResult != 0)
+            revert RedemptionErrors.USDCRedeemFailed(redeemResult);
+
+        uint256 usdcAmount = (ICERC20(CUSDC).exchangeRateStored() * _amount) /
+            1e20;
```

Comment:
Why `1e20` ? Can it be moved to constants?

Reply:
Because:
USDC 1e6 = (CUSDC 1e8 * exchangeRate 1e18) / 1ePRECISION
PRECISION = 20

yeah - should be in constants